operate/client/src/App/Processes/ListView/v2/Filters/tests/index.test.tsx

```diff
@@ -0,0 +1,545 @@
+    const {user} = render(<Filters />, {
+      wrapper: getWrapper(),
+    });
+    expect(screen.getByTestId('search')).toBeEmptyDOMElement();
+
+    await user.click(screen.getByRole('button', {name: 'More Filters'}));
+    await user.click(screen.getByText('Parent Process Instance Key'));
+    await user.type(
+      screen.getByLabelText(/^parent process instance key$/i),
+      'a',
+    );
+
+    expect(
+      await screen.findByText('Key has to be a 16 to 19 digit number'),
+    ).toBeInTheDocument();
+
+    expect(screen.getByTestId('search')).toBeEmptyDOMElement();
+    await user.click(screen.getByRole('button', {name: 'More Filters'}));
+    await user.click(screen.getByText('Operation Id'));
+    await removeOptionalFilter({user, screen, label: 'Operation Id'});
+
+    expect(screen.getByTestId('search')).toBeEmptyDOMElement();
+  });
+
+  it('should be able to submit form after deleting an invalid optional filter', async () => {
+    const {user} = render(<Filters />, {
+      wrapper: getWrapper('/?active=true&incidents=true'),
+    });
+    expect(screen.getByTestId('search')).toHaveTextContent(
+      /^\?active=true&incidents=true$/,
+    );
+
+    await user.click(screen.getByRole('button', {name: 'More Filters'}));
+    await user.click(screen.getByText('Parent Process Instance Key'));
+    await user.type(
+      screen.getByLabelText(/^parent process instance key$/i),
+      'a',
+    );
+
+    expect(
+      await screen.findByText('Key has to be a 16 to 19 digit number'),
+    ).toBeInTheDocument();
+
+    expect(screen.getByTestId('search')).toHaveTextContent(
+      /^\?active=true&incidents=true$/,
+    );
+
+    await removeOptionalFilter({
+      user,
+      screen,
+      label: 'Parent Process Instance Key',
+    });
+
+    await user.click(screen.getByRole('button', {name: 'More Filters'}));
+    await user.click(screen.getByText('Error Message'));
+    await user.type(screen.getByLabelText(/^error message$/i), 'test');
+
+    await waitFor(() =>
+      expect(screen.getByTestId('search')).toHaveTextContent(
+        /^\?active=true&incidents=true&errorMessage=test$/,
+      ),
+    );
+  });
+
+  it('Should order optional filters', async () => {
+    const {user} = render(<Filters />, {
+      wrapper: getWrapper('/?active=true&incidents=true'),
+    });
+
+    const optionalFilters: Array<{name: string; fields: string[]}> = [
+      {name: 'Error Message', fields: ['Error Message']},
+      {
+        name: 'Parent Process Instance Key',
+        fields: ['Parent Process Instance Key'],
+      },
+      {name: 'Variable', fields: ['Name', 'Value']},
+      {name: 'Process Instance Key(s)', fields: ['Process Instance Key(s)']},
+      {name: 'Operation Id', fields: ['Operation Id']},
+    ];
+
+    let fieldLabels = optionalFilters.reduce((acc, optionalFilter) => {
+      return [...acc, ...optionalFilter.fields];
+    }, [] as string[]);
+
+    for (let i = 0; i < optionalFilters.length; i++) {
```

Comment:
We can replace these for with indexes with [`for ... of`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of)